.github/workflows/docker.yml

```diff
@@ -1,163 +1,134 @@
+
+env:
+    # Enable self-hosted runners for the sigp repo only.
+    SELF_HOSTED_RUNNERS: ${{ github.repository == 'sigp/anchor' }}
+    RUST_VERSION: '1.80.0'
+    SHORT_ARCH: 'amd64'
+    # Deny warnings in CI
+    RUSTFLAGS: "-D warnings -C debuginfo=0"
+    # Prevent Github API rate limiting
+    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+    extract-version:
+        uses: ./.github/workflows/extract-version.yml
+
+    
+    build-docker-multi-arch:
+        name: build-docker-anchor-${{ matrix.cpu_arch }}
+        runs-on: ${{ github.repository == 'sigp/anchor' && fromJson('["self-hosted", "linux", "release"]') || 'ubuntu-22.04'  }}
+        strategy:
+            matrix:
+                cpu_arch: [aarch64, x86_64]
+                include:
+                    - profile: maxperf
+
+        needs: [extract-version]
+        env:
+            VERSION: ${{ needs.extract-version.outputs.VERSION }}
+            VERSION_SUFFIX: ${{ needs.extract-version.outputs.VERSION_SUFFIX }}
+        steps:
+            
+            - name: Map aarch64 to arm64 short arch
+              if: startsWith(matrix.cpu_arch, 'aarch64')
+              run: echo "SHORT_ARCH=arm64" >> $GITHUB_ENV
+
+            - name: Map x86_64 to amd64 short arch
+              if: startsWith(matrix.cpu_arch, 'x86_64')
+              run: echo "SHORT_ARCH=amd64" >> $GITHUB_ENV
+
+
+            - name: Install Rust and Cargo
+              run: |
+                curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+                source $HOME/.cargo/env
+                echo "PATH=$HOME/.cargo/bin:$PATH" >> $GITHUB_ENV
+
+                
+            - name: cargo install cross
+              run: |
+                    cargo install cross
+
+            # - uses: actions/checkout@v4
+            # - name: Update Rust
+            #   if: env.SELF_HOSTED_RUNNERS == 'false'
+            #   run: rustup update stable
+            - name: Checkout sources
+              uses: actions/checkout@v4
+
+            - name: Get rust-version 
+              id: get-rust-version 
+              run: |
+                echo "RUST_VERSION=$(./.github/scripts/toml_reader.sh ./anchor/Cargo.toml package rust-version)" >> $GITHUB_ENV
+          
+            - name: Get latest version of stable Rust
+              run: echo "rust version is ${{ env.RUST_VERSION }}"
+
+            - name: Retrieve Docker credentials from Vault
+              uses: hashicorp/vault-action@v2
+              with:
+                url: https://vault.sigp.io
+                method: github
+                githubToken: ${{ secrets.GH_TOKEN }}
+                secrets: |
+                    spesi_kv/data/dev/docker/anchor DOCKER_USERNAME ;
+                    spesi_kv/data/dev/docker/anchor DOCKER_PASSWORD
+
+            - name: Dockerhub login
+              run: |
+                  echo "${DOCKER_PASSWORD}" | docker login --username ${DOCKER_USERNAME} --password-stdin
```

Comment:
I think these have changed. They are now `DH_ORG` and `DH_KEY` I think

Reply:
In these workflows we are not trusting github to responsibility protect the dockerhub token in perpetuity. Instead we give it on demand, and approval, via vault. 
While it does use `DOCKER_USERNAME` and `DOCKER_PASSWORD` vars internally, the only variable that needs to be set in Github is to add your org token. See https://www.notion.so/sigp/HashiCorp-Vault-12a56eefb55e8023bb58de9d57cfe2b8